src/parser/jobs/rpr/modules/ArcaneCircle.tsx

```diff
@@ -38,6 +47,13 @@ export class ArcaneCircle extends BuffWindow {
 		}))
 	}
 
+	private maybeReOpenPreviousWindow(event: Events['statusApply']) {
+		// See if another reaper overrode your Arcane Circle
+		if ((this.history.endOfLastEntry() ?? 0) === event.timestamp) {
```

Comment:
I think this should be able to compare without the `??` and will return false if `endOfLastEntry` is undefined.

```suggestion
		if (this.history.endOfLastEntry() === event.timestamp) {
```